platform/windows/os_windows.cpp

```diff
@@ -290,6 +291,25 @@ String OS_Windows::get_name() const {
 	return "Windows";
 }
 
+String OS_Windows::get_distribution_name() const {
+	return get_name();
+}
+
+String OS_Windows::get_version() const {
+    DWORD majorVersion = 0;
+    DWORD minorVersion = 0;
+	DWORD build = 0; 
+	DWORD gversion = (DWORD)GetProcAddress(GetModuleHandle("kernel32.dll"), "GetVersion");
```

Comment:
GetVersion result might be changed by compatibility settings or app manifest, I would suggest using RtlGetVersion instead, https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion

Reply:
For macOS, you can use NSProcessInfo to get proper version, distribution name can be hardcoded to `macOS`:

```obj-c++
String OS_MacOS::get_version() const {
	NSOperatingSystemVersion ver = [NSProcessInfo processInfo].operatingSystemVersion;
	return vformat("%d.%d.%d", (int64_t)ver.majorVersion, (int64_t)ver.minorVersion, (int64_t)ver.patchVersion);
}
```

Reply:
I reckon that works also for iOS?